caddyhttp/status/status.go

```diff
@@ -0,0 +1,23 @@
+// Package status is middleware for returning status code for requests
+package status
+
+import (
+	"net/http"
+
+	"github.com/mholt/caddy/caddyhttp/httpserver"
+)
+
+// Status is a middleware to return status code for request
+type Status struct {
+	Rules map[string]int
+	Next  httpserver.Handler
+}
+
+// ServeHTTP implements the httpserver.Handler interface
+func (status Status) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+	if statusCode, ok := status.Rules[r.URL.Path]; ok {
+		return statusCode, nil
```

Comment:
Hmm, this middleware is an odd exception in the way I've described how the [contract](https://godoc.org/github.com/mholt/caddy/caddyhttp/httpserver#Handler) for a Handler works:

> If ServeHTTP writes to the response body, it should return a status code of 0. This signals to other handlers above it that the response body is already written, and that they should not write to it also.

In this case we don't write a response body but we do essentially "finish" handling the request (which is what I meant by "writes to the response body" - perhaps I should clarify that in the godoc by changing it to "writes the response header").

So what happens here is two-fold:

- If we return statusCode >= 400, great; an error page can be served by the errors middleware.
- If we return statusCode < 400... not so great, because there's no code to write a response body in that case, or even write the status code to the header of the response.

In other words, we need a condition, something like:

```go
if statusCode < 400 {
    w.WriteHeader(statusCode)
    return 0, nil
}
return statusCode, nil
```